File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    key_dict = {}
    for key in keymap:
        for idx, str in enumerate(key):
            if str not in key_dict:
                key_dict[str] = idx+1
            else:
                cur_val = key_dict[str]
                key_dict[str] = min(idx+1, cur_val)
                
    result = []
    for target in targets:
        target_sum = 0
        for s in target:
            if s not in key_dict: 
                result.append(-1) 
                break
            target_sum += key_dict[s]
        else:
            result.append(target_sum)
            
    return result
            

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
const solution = (keymap, targets) => {
    const key_dict = {}
    
    keymap.forEach((keys) => {
        for(let i=0; i<keys.length; i++) {
            const cur_key = keys[i];
            key_dict[cur_key] = Math.min(key_dict[cur_key] ?? i + 1, i + 1);
        }
    });
    
    result = []
    
    targets.forEach((target) => {
        let total_sum = 0
        let can_make = true
        
        for (const s of target) {
            if (key_dict[s] === undefined) {
                result.push(-1);
                can_make = false;
                break;
            } else {
                total_sum += key_dict[s];
            }
        }
        if (can_make) { result.push(total_sum); }
    });
    return result;
} 